modules/host/negotiatesettings.go

```diff
@@ -26,54 +26,60 @@ func (h *Host) capacity() (total, remaining uint64, err error) {
 	return total, remaining, nil
 }
 
+// externalSettings compiles and returns the external settings for the host.
+func (h *Host) externalSettings() (modules.HostExternalSettings, error) {
+	totalStorage, remainingStorage, err := h.capacity()
+	if err != nil {
+		return modules.HostExternalSettings{}, err
+	}
```

Comment:
Oh. So, there are actually a handful of errors here, but the biggest is that the call to `StorageFolders` returns more than capacity, it calls `filepath.EvalSymlinks` to return the the path of the storage folder. It also grabs a resource lock but I don't think it needs to do that, the function doesn't actually seem to use any resources that can close.

If we can figure out some other way to handle an error from `filepath.EvalSymlinks`, then we should be able to drop the error. Capacity itself is tracked in memory. I guess instead of calling EvalSymlinks during `StorageFolders()` I could store the path when the directory gets created.

I'm wondering if using symlinks was the right decision. In hindsight, it seems a little strange, but I do think that you get to keep the portability and everything. Right now, the host does all operations inside of its own folder, and when you add a storage folder it makes a symlink to the folder that you added, and then forgets that the original ever existed. Which is why you need to evaluate the symlinks to remember where the original storage folder was.

I suppose we could have the whole thing be in a .json file, but I think using symlinks made a lot of the operations a lot simpler. Plus, in a file manager, all of your host data is going to be in one place (via symlinks) which makes everything a lot easier to browse. Not that there's any reason to do that beyond natural curiosity, but still it's a nice bonus.